hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/HoodieFlinkWriteClient.java

```diff
@@ -94,7 +94,7 @@
    * FileID to write handle mapping in order to record the write handles for each file group,
    * so that we can append the mini-batch data buffer incrementally.
    */
-  private final Map<String, HoodieWriteHandle<?, ?, ?, ?>> bucketToHandles;
+  private final Map<String, MiniBatchHandle> bucketToHandles;
 
```

Comment:
Thanks, I'll try to rework it

Reply:
- Reverted replacing `HoodieWriteHandle` with `MiniBatchHandle`.
- Reverted `FlinkClosedHandle`
- Introduced class `BucketHandles` replacing with `Map<String, HoodieWriteHandle<?, ?, ?, ?>> bucketToHandles` to keep handles and write paths
- Added test `BucketHandles`

Workload benchmark results are the same

Reply:
We need to clarify whether `HoodieFlinkWriteClient#cleanHandlesGracefully` is still valid. I wrote the logic to want it work like post task killing hook, if a writing task was force killed when it is writing a file, how can we handle this more gracefully.

For most of the case, if the writing task was force killed or canceled, the file writer should had been closed right? I'm skeptical whether it is still needed.

It is related with this issue, because if `cleanHandlesGracefully` is not valid anymore, there is no need to cache any write handles at all.

Reply:
Thanks, that is a good point, I'll check it

Reply:
I think you are right. There is no need to store writers because they closed by `HoodieConsumer.finish()`. So if we have `List<WriteStatus> writeStatus` then the writer is already closed. We should keep only one `currentWriter` to handle exception during write operation. I'm preparing updated fix now